tools/ports/contrib/glfw3.py

```diff
@@ -4,23 +4,44 @@
 # found in the LICENSE file.
 
 import os
-
-TAG = '1.0.4'
-HASH = 'c3c96718e5d2b37df434a46c4a93ddfd9a768330d33f0d6ce2d08c139752894c2421cdd0fefb800fe41fafc2bbe58c8f22b8aa2849dc4fc6dde686037215cfad'
+from urllib.parse import parse_qs
 
 # contrib port information (required)
 URL = 'https://github.com/pongasoft/emscripten-glfw'
 DESCRIPTION = 'This project is an emscripten port of GLFW written in C++ for the web/webassembly platform'
 LICENSE = 'Apache 2.0 license'
 
+OPTIONS = {
+  'tag': 'The git tag to use a different release',
+  'hash': 'The sha512 of the release associated to the tag (can be omitted)',
+  'disableWarning': 'Boolean to disable warnings emitted by the library',
+  'disableJoystick': 'Boolean to disable support for joystick entirely',
+  'disableMultiWindow': 'Boolean to disable multi window support which makes the code smaller and faster'
+}
+
+# user options (from --use-port)
+opts = {
+  'tag': '1.0.4',
+  'hash': 'c3c96718e5d2b37df434a46c4a93ddfd9a768330d33f0d6ce2d08c139752894c2421cdd0fefb800fe41fafc2bbe58c8f22b8aa2849dc4fc6dde686037215cfad',
```

Comment:
Lets leave the tag and hash hardcoded for now if thats ok.

Reply:
I understand your point of view. I will revert this change.